Sources/ApolloSQLite/SQLiteNormalizedCache.swift

```diff
@@ -33,11 +33,19 @@ public final class SQLiteNormalizedCache {
   }
 
   private func recordCacheKey(forFieldCacheKey fieldCacheKey: CacheKey) -> CacheKey {
-    var components = fieldCacheKey.components(separatedBy: ".")
-    if components.count > 1 {
-      components.removeLast()
+    let components = fieldCacheKey.components(separatedBy: ".")
+    var updatedComponents = [String]()
+    for component in components {
+      if updatedComponents.last?.last?.isNumber ?? false && component.first?.isNumber ?? false {
+        updatedComponents[updatedComponents.count - 1].append(".\(component)")
+      } else {
+        updatedComponents.append(component)
+      }
+    }
```

Comment:
Does this need to loop through every single component, or does it only need to look at the last couple of components? I'm concerned if we loop through every single component every single time, that's going to wind up chewing up a bunch of energy we may not need to.

Reply:
I revisited this, need to loop through but only if we see it's a "_ROOT" so it shouldn't happen very often.